src/download-database.ts

```diff
@@ -0,0 +1,187 @@
+// This file borrows heavily from the actions "downloadTool" function:
+// https://github.com/actions/toolkit/blob/main/packages/tool-cache/src/tool-cache.ts#L38
+
+import * as fs from "fs";
+import * as path from "path";
+import * as stream from "stream";
+import * as util from "util";
+
+import * as core from "@actions/core";
+import * as httpm from "@actions/http-client";
+import { IHeaders } from "@actions/http-client/interfaces";
+import * as io from "@actions/io";
+
+export class HTTPError extends Error {
+  httpStatusCode: number | undefined;
+  httpMessage: string;
+  constructor(httpStatusCode: number | undefined, httpMessage: string) {
+    super(`Unexpected HTTP response: ${httpStatusCode}. ${httpMessage}`);
+    // Set status code and error message to `this`
```

Comment:
```suggestion
```
I'm not sure this comment makes much sense and I suggest getting rid of it. If I remember rightly we just added it in the hope that copilot would do something useful. If you do want to keep it then I think it would make more sense as "set status code and error message on this" instead of "to this".